### DISCUSSION TITLE
posts/Go/var-and-const/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#355] [Announcements]
https://www.liwenzhou.com/posts/Go/01_var_and_const/ 

 李文周的Blog

**Comment by chengkanghua** [comment]
定义数量级有疑问？
```
定义数量级
const (
		_  = iota   # 这里用的匿名变量，默认值 0开始
		KB = 1 << (10 * iota)  #这里 iota 数值是1 10*1 =10 ，  << 这个符号啥意思？
		MB = 1 << (10 * iota)
		GB = 1 << (10 * iota)
		TB = 1 << (10 * iota)
		PB = 1 << (10 * iota)
	)
```

**Comment by ceily** [comment]
> @chengkanghua
> 定义数量级有疑问？
> ```
> 定义数量级
> const (
> 		_  = iota   # 这里用的匿名变量，默认值 0开始
> 		KB = 1 << (10 * iota)  #这里 iota 数值是1 10*1 =10 ，  << 这个符号啥意思？
> 		MB = 1 << (10 * iota)
> 		GB = 1 << (10 * iota)
> 		TB = 1 << (10 * iota)
> 		PB = 1 << (10 * iota)
> 	)
> ```

**Reply by ZYWNB666** [reply]
二进制左移位数，1的二进制是1，左移10位得到10000000000，换成十进制就是1024

**Comment by ceily** [comment]
乘以2的n次方，即1*2的10次方等于1024

**Comment by ChillyYep** [comment]
> @chengkanghua
> 定义数量级有疑问？
> ```
> 定义数量级
> const (
> 		_  = iota   # 这里用的匿名变量，默认值 0开始
> 		KB = 1 << (10 * iota)  #这里 iota 数值是1 10*1 =10 ，  << 这个符号啥意思？
> 		MB = 1 << (10 * iota)
> 		GB = 1 << (10 * iota)
> 		TB = 1 << (10 * iota)
> 		PB = 1 << (10 * iota)
> 	)
> ```

这个是位运算，"KB = 1 << (10 * iota)"的意思就是将1左移10位。如果1是16位整型，二进制为00000000 00000001，左移10位后变成00000100 00000000。左移的话，如果学过计算机组成原理应该知道有几种左移的方法，这里的左移策略应该是左边超出部分会被扔掉，右边不足的部分用0来填充。所以就16位整型1左移10位的结果来说，就是2^10。

**Comment by cg430** [comment]
打卡

**Comment by sealevelsun** [comment]
支持大佬

**Comment by SmeagolYe** [comment]
done

**Comment by GoHelloWorld** [comment]
宝藏博主

**Comment by kokosay** [comment]
打卡

**Comment by wxweven** [comment]
const (
		_  = iota
		KB = 1 << (10 * iota)
		MB = 1 << (10 * iota)
		GB = 1 << (10 * iota)
		TB = 1 << (10 * iota)
		PB = 1 << (10 * iota)
	)

这种写法，感觉有『炫技』的嫌疑，增加了不少的理解程度，像下面这样写，不是简单又好懂吗？

const (
	_  = iota
	KB = 1 << (10 * iota)
	MB = (1<<10) * KB
	GB = (1<<10) * MB
	TB = (1<<10) * GB
	PB = (1<<10) * TB
)

**Comment by ZYWNB666** [comment]
打卡
